utility/src/main/java/org/oppia/util/parser/LiTagHandler.kt

```diff
@@ -1,37 +1,171 @@
 package org.oppia.util.parser
 
+import android.content.Context
 import android.text.Editable
 import android.text.Html
 import android.text.Spannable
 import android.text.Spanned
-import android.text.style.BulletSpan
+import android.text.Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+import android.text.Spanned.SPAN_MARK_MARK
+import android.text.style.LeadingMarginSpan
+import java.util.Stack
+import org.oppia.util.parser.LiTagHandler.ListTag
+import org.oppia.util.parser.StringUtils.LI_TAG
+import org.oppia.util.parser.StringUtils.OL_TAG
+import org.oppia.util.parser.StringUtils.UL_TAG
 import org.xml.sax.XMLReader
 
 /**
- * [Html.TagHandler] implementation that processes <li> tags and creates bullets.
+ * Called when the HTML parser reaches an opening or closing tag.
+ * We only handle list tags and ignore other tags.
  *
- * Reference: https://github.com/davidbilik/bullet-span-sample
+ * <ul> and <ol> tags are pushed to the [lists] stack and popped when the closing tag is reached.
+ *
+ * <li> tags are handled by the [ListTag] instance corresponding to the parent tag.
+ *
+ * Reference: https://github.com/daphliu/android-spannable-list-sample/tree/master/app/src/main/java/com/daphneliu/sample/listspansample
  */
-class LiTagHandler : Html.TagHandler {
+class LiTagHandler(private val context: Context) : Html.TagHandler {
```

Comment:
Can you also add a test for a string where the first part of the string is a list with no tags before it? I recall there being a bug in Android's custom tag handler integration where it won't properly call into handleTag() for the first custom tag it encounters.

Reply:
I didnt get it correctly. Can you give me an example for this please.

Reply:
Hmm, I think the example you're using actually satisfies this. I'm not sure why that bug isn't being found here.